### PR TITLE
Added static unroll limit of 25; added test

### PR DESCRIPTION
**Pull request by rybern**
Previously, the static loop unrolling optimization would unroll constant size loops of any size, which could lead to code size issues. This PR limits the number of unrolls to an arbitrary 25.

### CODE REVIEW

File: test/integration/good/compiler-optimizations/unroll-limit.stan [hunk 1]
```diff
@@ -0,0 +1,9 @@
+generated quantities {
+  int x;
+  for (i in 1:100) {
+    x += 1;
+  }
+  for (i in 1:20) {
```
Comment (rok-cesnovar) [inline test/integration/good/compiler-optimizations/unroll-limit.stan:6]: Mind just adding a test that tests that low/upp work, so something from 50:70 and 50 to 100?

Comment (rok-cesnovar) [reply test/integration/good/compiler-optimizations/unroll-limit.stan:6]: Also from -10 to 10 and -10 to 50 for example.

Comment (rybern) [reply test/integration/good/compiler-optimizations/unroll-limit.stan:6]: Good idea, adding that

**Comment by SteveBronder** [comment]
Should it be a multiple of 2? Like 16 or 32?

**Comment by rok-cesnovar** [comment]
I am fine with 32.

**Comment by rok-cesnovar** [comment]
Feel free to merge once tests pass.

**Pull request merged**